pulp_rpm/app/downloaders.py

```diff
@@ -58,6 +63,19 @@ def __init__(
 
         super().__init__(*args, **kwargs)
 
+        if urlencode:
```

Comment:
Not thrilled at all with this solution, we're kind of piling hacks on top of hacks.  But there's so many different cases that need to be handled that I'm not sure which alternative ways of handling it might work.

Reply:
imnsho, the root problem is that mirrorlist-urls **are not** much like remote-urls, and overloading "url" the way we do is asking for pain. However, I don't think that level of refactoring is in the cards. Given the path we're on, lots of comments and trying to keep the special-casing in a small number of places is probably the best we can realistically manage.